include/path_tracking_pid/details/fifo_array.hpp

```diff
@@ -24,6 +24,9 @@ class FifoArray
   // Read-only access to the element at the given index.
   constexpr const value_type & operator[](std::size_t index) const { return data_[index]; }
 
+  // Read-write access to the element at the given index.
+  value_type & operator[](std::size_t index) { return data_[index]; }
```

Comment:
As far as I can tell you only use this functionality in `SecondOrderLowpass` because you want a calculation to "look" correct. Basically you push a dummy value and then overwrite it. I don't think that justifies adding this function. Just my 2 cents. If you decide to keep it, add some unit tests for it.

Reply:
You are correct, technically it is not needed. But it is very helpful if the math corresponds to the code. I'll add some tests for this